website/project/model.py

```diff
@@ -309,6 +334,17 @@ def edit(self, content, auth, save=False):
         self.content = content
         self.modified = True
         self.date_modified = datetime.datetime.utcnow()
+
+        if (new_mentions):
+            self.new_mentions = new_mentions
+            old_mentions = self.old_mentions
+            validate = lambda m: m not in old_mentions and validate_contributor(m, self.node.contributors)
+            self.new_mentions = filter(validate, self.new_mentions)
+            if len(self.new_mentions) > 0:
+                if save:
+                    project_signals.mention_added.send(self, auth=auth)
+                    self.old_mentions.extend(self.new_mentions)
```

Comment:
> Perhaps we factor out a function to DRY up with the create case

Done in ee9fad71a7420512f1a56a735344a9964d47841a.

> [W]e don't modify [`ever_mentioned`] except when we save. Seems like we should fix that.

I have a recollection that we decided that this is the correct behavior here, after all. Sound familiar, @leb2dg?